app/hooks/useAPI.tsx

```diff
@@ -0,0 +1,37 @@
+'use client'
+import { SUDTApi, APIClientOptions } from '@/app/components/apiClient'
+import { useContext, createContext, useState, useEffect, PropsWithChildren } from 'react'
+
+export interface IAPIContext {
+  apiClient: SUDTApi,
+}
+
+export const APIContext = createContext<IAPIContext | undefined>(undefined);
+
+export const useAPIClient = (): IAPIContext => {
+  const apiContext = useContext(APIContext);
+  if (!apiContext) {
+    throw new Error(
+        'No APIContext.Provider found when calling useAPIClient.'
+    );
+  }
+  return apiContext;
+};
+
+interface APIClientProviderProps {
+  options?: APIClientOptions
+}
+
+export const APIClientProvider = ({ children, options: defaultOptions }: PropsWithChildren<APIClientProviderProps>) => {
+  const options = {
+    origin: process.env.NEXT_PUBLIC_API_ENDPOINT || '/',
+    ...defaultOptions,
+  }
+  const [client, setClient] = useState(new SUDTApi(options));
+
+  return (
+    <APIContext.Provider value={{ apiClient: client }}>
```

Comment:
Is it necessary to inject `client` into `context`?

Reply:
The expected usage is that other components  called via the `const { apiClient } = useAPI()` to get an instance of the API
Are there any side effects to injecting into the `context`?

Reply:
The boundary becomes obscure because `api` is unrelated to `state`. It can be shared as a singleton. IMO, only global states should be shared in context

Reply:
It makes sense.